source/parsing/Parser_members.cpp

```diff
@@ -3227,7 +3227,7 @@ SpecparamDeclaratorSyntax& Parser::parseSpecparamDeclarator(SyntaxKind parentKin
             if (parentKind != SyntaxKind::SpecifyBlock)
                 addDiag(diag::PulseControlSpecifyParent, name.range());
             else if (!expr2)
-                addDiag(diag::PulseControlTwoValues, expr1.sourceRange()) << name.range();
+                expr2 = &expr1;
```

Comment:
This should not duplicate the expression here -- leaving it null seems fine. The parse tree must maintain the property that visiting the tokens in it via depth-first traversal recovers the original source file as typed by the user.

Reply:
fxied in last commit with simple code refactoring